src/js/view/preview.js

```diff
@@ -46,16 +45,48 @@ define([
             }
         },
         setImage: function(img) {
+            // Remove onload function from previous image
+            var image = this.image;
+            if (image) {
+                image.onload = null;
+                this.image = null;
+            }
             this.model.off('change:state', null, this);
             var backgroundImage = '';
             if (_.isString(img)) {
                 backgroundImage = 'url("' + img + '")';
+                image = this.image = new Image();
+                image.src = img;
             }
             utils.style(this.el, {
                 backgroundImage: backgroundImage
             });
         },
-        element : function() {
+        resize: function(width, height, stretching) {
+            if (stretching === 'uniform') {
+                var playerAspectRatio = width / height;
+                // snap image to edges when the difference in aspect ratio is less than 9%
+                var image = this.image;
+                var backgroundSize = null;
+                if (image) {
+                    if (image.width === 0) {
+                        var _this = this;
+                        image.onload = function() {
+                            _this.resize(width, height, stretching);
```

Comment:
@pajong by moving the resize code into preview, we now retry on image load when the image is ready instead of using a Promise.